pkg/server/server.go

```diff
@@ -153,6 +156,25 @@ func (pm *PendingDialManager) Remove(random int64) *ProxyClientConnection {
 	return pd
 }
 
+func (pm *PendingDialManager) RemoveForStream(streamUID string) []*ProxyClientConnection {
```

Comment:
Would be nice to have this documented. Especially that the result is a slice of the removed streams.

Reply:
Also why are we making this public?

Reply:
> Would be nice to have this documented.

Done.

> Also why are we making this public?

I did it for consistency but agree it can be kept private; done.